From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Evan Stade <evanstade@microsoft.com>
Date: Fri, 11 Sep 2026 16:11:38 -0700
Subject: Convert DCHECK before downcasting DOMArrayBufferBase to CHECK

There are certainly other DCHECKs in this general area which may also be
better as CHECKs, but they're less obviously problematic, so they are
likely better off going through the NotFatalUntil pipeline.

Bug: 558764482
Change-Id: Ia1e9e893fb419c5e346276933dc5ea4e132a1122
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/8395471
Commit-Queue: Evan Stade <evanstade@microsoft.com>
Reviewed-by: Daniel Cheng <dcheng@chromium.org>
Cr-Commit-Position: refs/heads/main@{#1696699}

diff --git a/third_party/blink/renderer/core/typed_arrays/dom_array_buffer_view.h b/third_party/blink/renderer/core/typed_arrays/dom_array_buffer_view.h
index 357602163e3b99dac9e226392fd11804ceb5990d..e20f3bc07c5dece9858cd849f3674e132c1293e9 100644
--- a/third_party/blink/renderer/core/typed_arrays/dom_array_buffer_view.h
+++ b/third_party/blink/renderer/core/typed_arrays/dom_array_buffer_view.h
@@ -40,13 +40,13 @@ class CORE_EXPORT DOMArrayBufferView : public ScriptWrappable {
   ~DOMArrayBufferView() override = default;
 
   DOMArrayBuffer* buffer() const {
-    DCHECK(!IsShared());
+    CHECK(!IsShared());
     DCHECK(dom_array_buffer_);
     return static_cast<DOMArrayBuffer*>(dom_array_buffer_.Get());
   }
 
   DOMSharedArrayBuffer* BufferShared() const {
-    DCHECK(IsShared());
+    CHECK(IsShared());
     DCHECK(dom_array_buffer_);
     return static_cast<DOMSharedArrayBuffer*>(dom_array_buffer_.Get());
   }
